src/components/TodoComponents/TodoForm.js

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+function ToDoForm(props) {
+    return (
+        <div>
+            <form>
+                <input type="text" name="todo" placeholder="...todo" value={props.todo} onChange={props.onChange}/>
+                <button type="submit" onSubmit={props.submitHandler}>Add Todo</button>
```

Comment:
I'm not entirely sure why onSubmit is not working here, but using an `onClick` instead worked for me along with other small changes to the App component.